### DISCUSSION TITLE
Restarting an interrupted large borg extract

### DISCUSSION THREAD

**Discussion by dokbua** [borgbackup/borg#6654] [Q&A] [answered]
Say you need to restore a large'ish, as in terabytes, failed disk drive.  It seems that _borg extract_ is supposed to be faster than  _borg mount_ + rsync, but what if something interrupts the restore halfway through - like for example a faulty internet connection or something.  What is the best way to restart the process?   

[The docs](https://borgbackup.readthedocs.io/en/stable/usage/extract.html) says _borg extract_ should be done into an empty directory, so that would require a restart from scratch?  I imagine restarting from where it got interrupted would cause it to extract and overwrite the already extracted files, which could take a long time, particularly over an internet connection.  

How much slower would rsync from a _borg mount_ be?  

What is the best solution, keeping in mind that a restore in this case would be considered an emergency?

**Comment by ThomasWaldmann** [comment, marked as answer]
borg extract has only minimal support for restoring into a non-empty directory (like e.g. if a path it wants to extract already exists, it would delete the current file first to get it out of the way and then extract from the archive). do not expect anything more complex than that.

if you have a rather unstable connection so that an interruption is likely, you might be better off with borg mount and rsync, just be aware that borg mount (the FUSE fs) does not support all metadata as borg extract does, esp. no ACLs, no (bsd)flags, no xattrs(?).

an alternative might be doing multiple well-defined and distinct partial extracts (see borg extract docs), like e.g. extracting /home/user1 first, then /home/user2.

i can't say how much slower borg mount would be for you, I've never compared them and even if I had, there are a lots of parameters influencing performance.
